Assets/Scenes/50-Minigames/512-PlaneGame/Scripts/FlightControls.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor.Rendering;
+using UnityEngine;
+
+public class NewBehaviourScript : MonoBehaviour
+{
+
+    [SerializeField]
+    private GameObject playerObject;
+
+    private float speed = 5;
+    private float rotationDuration = 2f;
+    private bool isSpinning = false;
+    
+
+
+    void Update()
+    {
+        playerFlight();
+    }
+
+
+    /// <summary>
+    /// diffrent direction the player charector can fly With WASD buttons, aswell as barrel roll via Spacebar.
+    /// </summary>
+    private void playerFlight()
+    {
+       
+        if (Input.GetKey(KeyCode.W))
+        {
+            playerObject.transform.position += new Vector3(0, speed * Time.deltaTime, 0);
+   
+        }
+        if (Input.GetKey(KeyCode.S))
+        {
+            playerObject.transform.position -= new Vector3(0, speed * Time.deltaTime, 0);
+
+        }
+
+
+        if (Input.GetKey(KeyCode.A))
+        {
+            playerObject.transform.position -= new Vector3(speed * Time.deltaTime, 0, 0);
+
+        }
+        if (Input.GetKey(KeyCode.D))
+        {
+            playerObject.transform.position += new Vector3(speed * Time.deltaTime, 0, 0);
+
+        }
```

Comment:
i like this, but you dont need to inplement it if you dont understand it or dont like this code
` Transform transform = playerObject.transform;
+        Vector3 movement = new Vector3(
+            Input.GetAxis("Horizontal"),
+            Input.GetAxis("Vertical"),
+            0
+        ) * speed * Time.deltaTime;
+
+        Vector3 newPosition = transform.position + movement;`